src/tools.ts

```diff
@@ -116,6 +117,16 @@ export default class Tools {
         new Notice("Advanced URI copied to your clipboard");
     }
 
+    async copyMarkdownLink(parameters: Parameters) {
+        const markdownTitle = parameters.markdownTitle ? parameters.markdownTitle : "";
+        const uri = await this.generateURI(parameters, true);
+        const markdownLink = "[" + markdownTitle + "](" + uri + ")";
+
+        new Notice("Advanced URI copied to your clipboard (Markdown)");
```

Comment:
In case the copying fails, the notice should be placed after the actual copy function call.